### PR TITLE
use fopen directly when reading objects from s3

### PR DESCRIPTION
**Pull request by icewind1991**
Bypassing curl/guzzle.

This *might* help with errors when dealing with large downloads

### CODE REVIEW

**Review by MorrisJobke — APPROVED** [review]
Tested and at least works with the existing files I have in my test environment.

File: lib/private/Files/ObjectStore/S3ObjectTrait.php [hunk 1]
```diff
@@ -49,12 +49,15 @@ function readObject($urn) {
 			'Bucket' => $this->bucket,
 			'Key' => $urn
 		]);
-		$command['@http']['stream'] = true;
-		$result = $client->execute($command);
-		/** @var StreamInterface $body */
-		$body = $result['Body'];
+		$request = \Aws\serialize($command);
+		$opts = array(
```
Comment (MorrisJobke) [inline lib/private/Files/ObjectStore/S3ObjectTrait.php]: Could you use `[]` instead of `array()`

Comment (icewind1991) [reply lib/private/Files/ObjectStore/S3ObjectTrait.php]: done

**Comment by MorrisJobke** [comment]
Does a backport make sense here?

**Comment by schiessle** [comment]
@MorrisJobke yes, we need to backort it, please review

**Pull request merged**